src/containers/Login/index.js

```diff
@@ -0,0 +1,108 @@
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { LogInRequest } from '../../redux/actions/login';
+import { SocialButtons } from '../../components/SocialButtons';
+import { Input } from '../../components/elements';
+import './style.sass';
+
+class Loginform extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      password: '',
+      errors: {},
+    };
+    this.onChange = this.onChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+
+  onChange = (e) => {
+    // console.log('changed', e.target.name);
+    const { name, value } = e.target;
+    const { errors } = this.state;
+    delete errors[name];
+    this.setState({
+      [name]: value,
+      errors,
+    });
+  };
+
+  renderMessage = () => (
+    <div className={this.props.success ? 'ui floating message success' : 'ui floating message error'}>
+      { this.props.message }
+    </div>
+  )
+
+  onSubmit = (e) => {
+    e.preventDefault();
+    // console.log('this is it');
+    const { signin } = this.props;
+    signin({ ...this.state });
+  }
+
+  render() {
+    const { email, password } = this.state;
+    const loginInputs = [
+      {
+        name: 'email',
+        value: email,
+        id: 'email',
+        type: 'text',
+        icon: 'mail outline ',
+        placeholder: 'Enter email address',
+        onChange: this.onChange,
+      },
+      {
+        name: 'password',
+        value: password,
+        id: 'password',
+        type: 'password',
+        icon: 'lock',
+        placeholder: 'Enter password',
+        onChange: this.onChange,
+      },
+    ];
+    return (
+      <div className="ui raised very padded center aligned text container segment container-main">
```

Comment:
i think this `jsx` can be abstracted to a method, just like what you have done to `renderMessage` or create a component and pass the necessary props to it then render the component here